internal/util/logging/handler_test.go

```diff
@@ -72,6 +75,12 @@ func TestHandler(t *testing.T) {
 			l := WithName(slog.New(h), "test.logger")
 			require.NoError(t, l.Handler().Handle(ctx, r))
 
+			// mongo handler doesn't preserve the order of attributes
```

Comment:
Well, but does it keep it stable? I find it strange that we text exact values for the JSON format but not for the mongo format that uses the same JSON primitives under the hood.

Reply:
Mongo handler doesn't keep the order of attributes stable. 

As the attributes are stored in map, the ordering is random.

Both `json`, and `console` handlers use `encoding/json`, which sorts attribute map keys lexicographically.
The `mongo` handler uses `bson.MarshalExtJSON` function, which doesn't perform any sorting.

Reply:
> As the attributes are stored in map, the ordering is random.

Could we use `bson.D`?

Reply:
Our logging handlers use [shared function](https://github.com/FerretDB/FerretDB/pull/4716/files#diff-ed4f1a531b6586a30017cb944e778ad59d2e61bfc4119017ae95334da77a7bc1R35) that returns attributes in map.

Changing `mongoLog`'s field from map to `bson.D` wouldn't change anything.

We could add additional logic that would iterate through map, create new `bson.D` object with sorted fields, and then pass it to `bson.MarshalExtJSON`, but I don't see why would we want that. As far as I'm aware the attribute ordering is not expected anywhere, besides the test case.

Reply:
Usage of `map` in `attrs(..)` function fully internal, wouldn't replacing with `bson.D` allow preserve the order for all our custom handlers?

I don't think we need to iterate the `map` and then sort it, we can let it have the order defined by the logger.

Reply:
I'm not sure if using `bson.D` internally is something that we want to do. The `attrs` function is meant to be used by our logging handlers. In that case for `consoleHandler`, we'd need to take this bson.D output and convert it to value that'd be able to be consumed by `encoding/json` encoder.

Reply:
> Our logging handlers use shared function

Then we probably should un-share it. Or maybe generalize it by using generics and/or iterators (if that would result in simpler code).
Having attributes in a stable order seems to be useful properly for logs – it makes them more readable, which is kind of important.
Let's to that in a separate PR, but in this issue.